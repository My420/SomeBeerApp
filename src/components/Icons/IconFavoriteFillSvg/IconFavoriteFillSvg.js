import React from 'react';
import PropTypes from 'prop-types';
import styles from './IconFavoriteFillSvg.module.scss';

const IconFavoriteFillSvg = ({ containerClass, pathClass }) => {
  return (
    <span className={containerClass}>
      <svg width="100%" height="100%" viewBox="0 0 24 24">
        <path
          className={pathClass}
          d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
        />
      </svg>
    </span>
  );
};

IconFavoriteFillSvg.propTypes = {
  containerClass: PropTypes.string,
  pathClass: PropTypes.string
};

IconFavoriteFillSvg.defaultProps = {
  containerClass: styles.container,
  pathClass: styles.path
};

export default IconFavoriteFillSvg;
