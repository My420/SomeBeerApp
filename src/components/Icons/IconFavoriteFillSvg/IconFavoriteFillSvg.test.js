import React from 'react';
import { shallow } from 'enzyme';
import IconFavoriteFillSvg from './IconFavoriteFillSvg';

describe('Test <IconFavoriteFillSvg /> component', () => {
  describe('Test <IconFavoriteFillSvg /> without props', () => {
    const wrapper = shallow(<IconFavoriteFillSvg />);

    test('should correct render component without props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  render span with default className', () => {
      expect(wrapper.find('span').prop('className')).toBe('container');
    });

    test('should  render svg inside span', () => {
      expect(wrapper.find('span.container').find('svg')).toHaveLength(1);
    });

    test('svg should  have path with default className', () => {
      expect(wrapper.find('path').prop('className')).toBe('path');
    });
  });

  describe('Test <IconFavoriteFillSvg /> with props', () => {
    const wrapper = shallow(
      <IconFavoriteFillSvg
        containerClass="testContainer"
        pathClass="testPath"
      />
    );

    test('should correct render component with props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should  correct set className at span ', () => {
      expect(wrapper.find('span').prop('className')).toBe('testContainer');
    });

    test('should  correct set className at path', () => {
      expect(wrapper.find('path').prop('className')).toBe('testPath');
    });
  });
});
