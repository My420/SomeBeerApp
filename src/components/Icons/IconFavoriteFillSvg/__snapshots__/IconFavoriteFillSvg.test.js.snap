// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <IconFavoriteFillSvg /> component Test <IconFavoriteFillSvg /> with props should correct render component with props 1`] = `
<span
  className="testContainer"
>
  <svg
    height="100%"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      className="testPath"
      d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
    />
  </svg>
</span>
`;

exports[`Test <IconFavoriteFillSvg /> component Test <IconFavoriteFillSvg /> without props should correct render component without props 1`] = `
<span
  className="container"
>
  <svg
    height="100%"
    viewBox="0 0 24 24"
    width="100%"
  >
    <path
      className="path"
      d="M12 4.248c-3.148-5.402-12-3.825-12 2.944 0 4.661 5.571 9.427 12 15.808 6.43-6.381 12-11.147 12-15.808 0-6.792-8.875-8.306-12-2.944z"
    />
  </svg>
</span>
`;
